Rubberduck.Refactorings/Abstract/IBaseRefactoring.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Rubberduck.Parsing.Rewriter;
+
+namespace Rubberduck.Refactorings
+{
+    /// <summary>
+    /// The heart of a refactoring: this part of the refactoring performs the actual transformation of the code once all necessary information has been gathered.
+    /// </summary>
+    /// <typeparam name="TModel">The model used by the refactoring containing all information needed to specify what to do.</typeparam>
+    public interface IBaseRefactoring<in TModel> where TModel : class, IRefactoringModel
```

Comment:
I feel that this really ought to be just `IRefactoring`. The original `IRefactoring` should be the one that gets renamed into `IInteractiveRefactoring`. That would also avoid the need to name the implementation `BaseRefactoringBase` which is kind of confusing, considering that we already have a `InteractiveRefactoringBase` that derives from the original `IRefactoring` interface. 

The current `RefactoringBase` should be probably renamed into `SelectionRefactoringBase`. 

I realize this is a fair amount of naming but I think it is necessary to keep it the type hierarchy clear & intuitive since the refactoring objects are quite complex in themselves and we want to avoid muddying the water due to inconsistent naming.

Reply:
I think `IInteractiveRefactoring` does not make sense for the old `IRefactoring` since not all implementations are supposed to be interactive. It really is an interface for all user-facing fully implemented refactorings.